rust_src/src/vectors.rs

```diff
@@ -139,25 +139,28 @@ impl LispVectorRef {
     pub fn as_slice(&self) -> &[LispObject] {
         unsafe {
             slice::from_raw_parts(
-                mem::transmute::<_, *const LispObject>(&self.contents),
+                &self.contents as *const [i64; 1] as *const LispObject,
+                //mem::transmute::<_, *const LispObject>(&self.contents),
                 self.len(),
             )
         }
     }
 
     #[inline]
-    pub fn as_mut_slice(&self) -> &mut [LispObject] {
+    pub fn as_mut_slice(&mut self) -> &mut [LispObject] {
         unsafe {
             slice::from_raw_parts_mut(
-                mem::transmute::<_, *mut LispObject>(&self.contents),
+                &self.contents as *const [i64; 1] as *mut LispObject,
+                //mem::transmute::<_, *mut LispObject>(&self.contents),
                 self.len(),
             )
         }
     }
 
     #[inline]
     pub unsafe fn get_unchecked(&self, idx: ptrdiff_t) -> LispObject {
-        ptr::read(mem::transmute::<_, *const LispObject>(&self.contents).offset(idx))
+        let tmp = &self.contents as *const [i64; 1] as *const LispObject;
```

Comment:
i64 here too: does this mean clippy was misleading? If so, we should report a bug upstream as I'm sure they'd be interested :)

Reply:
Clippy appears confused by Lisp_Object. It appears to give us the final base Rust type `i64` instead of `EmacsInt` or `Lisp_Object`. I am working through making sure all of the naked i64s introduced are properly named.

Reply:
I will see about making a small test case and checking with the Clippy folk.